extensions/native/recursion/src/stark/mod.rs

```diff
@@ -468,56 +498,60 @@ where
         }
 
         // 3. After challenge: one per phase
-        builder
-            .range(0, num_phases)
-            .for_each(|phase_idx_vec, builder| {
-                let phase_idx = phase_idx_vec[0];
-                let values_per_mat = builder.get(&opening.values.after_challenge, phase_idx);
-                let batch_commit = builder.get(after_challenge_commits, phase_idx);
-
-                let mat_idx: Usize<_> = builder.eval(RVar::zero());
-                let mats: Array<_, TwoAdicPcsMatsVariable<_>> = builder.array(values_per_mat.len());
-                builder.range(0, num_airs).for_each(|i_vec, builder| {
-                    let i = i_vec[0];
-                    let advice = builder.get(&m_advice_var.per_air, i);
-                    builder
-                        .if_ne(advice.num_challenges_to_sample.len(), RVar::zero())
-                        .then(|builder| {
-                            // Only 1 phase is supported.
-                            builder.assert_eq::<Usize<_>>(
-                                advice.num_challenges_to_sample.len(),
-                                RVar::one(),
-                            );
-                            let domain = builder.get(&domains, i);
-                            let trace_points = builder.get(&trace_points_per_domain, i);
-
-                            let after_challenge = builder.get(&values_per_mat, mat_idx.clone());
-
-                            let values = builder.array::<Array<C, _>>(2);
-                            builder.set_value(&values, 0, after_challenge.local);
-                            builder.set_value(&values, 1, after_challenge.next);
-                            let after_challenge_mat = TwoAdicPcsMatsVariable::<C> {
-                                domain,
-                                values,
-                                points: trace_points,
-                            };
-                            builder.set_value(&mats, mat_idx.clone(), after_challenge_mat);
-                            builder.inc(&mat_idx);
-                        });
-                });
-                builder.assert_eq::<Usize<_>>(mat_idx, values_per_mat.len());
+        builder.assert_eq::<Usize<_>>(opening.values.after_challenge.len(), num_phases);
+        builder.if_eq(num_phases, RVar::one()).then(|builder| {
```

Comment:
what is this doing? are you requiring there to be one challenge phase?

Reply:
We already require there to be at most one challenge phase in many places. This is just making it more explicit.